src/index.js

```diff
@@ -875,7 +875,7 @@ class ServerlessAppsyncPlugin {
           Type: 'AWS::AppSync::FunctionConfiguration',
           Properties: {
             ApiId: { 'Fn::GetAtt': [logicalIdGraphQLApi, 'ApiId'] },
-            Name: tpl.name,
+            Name: this.getCfnName(tpl.name),
```

Comment:
I saw in the discuss thread that there was some concerns around using `getCfnName`.
Alternatively, sanitization could be done using [`_.camelCase `](https://lodash.com/docs/4.17.15#camelCase) or [`_.snakeCase`](https://lodash.com/docs/4.17.15#snakeCase)